Upgrade reqwest to v0.10 to parepare async/.await

tetsuharuohzeki opened the pull request:
reqwest v0.9 exposed blocking API as `reqwest::*`.
However, v0.10 expose async API by default and blocking APIs are exposed
as `reqwest::blocking`.

By this change, I migrated code to `reqwest::blocking`.

If we change our codes to async/.await, I feel we should do it in
an another changeset because async-fy might require large design change.

## Related Issues

* https://github.com/rust-lang/rustup/pull/2112

tetsuharuohzeki commented:
Oops, I accidentally removed "draft" flag for this pull request.
We should wait the stable release for _reqwest_.

tesuji commented on download/Cargo.toml:
```diff
@@ -16,11 +16,11 @@ reqwest-backend = ["reqwest", "env_proxy", "lazy_static"]
+reqwest = { git = "https://github.com/seanmonstar/reqwest", rev = "7631c0390e90d539b825ed27693a7a0bb1e31dc8", features = ["blocking", "gzip"], optional = true }
 
 [dev-dependencies]
 hyper = "0.13"
```
```suggestion
hyper = { version = "0.13", default-features = false, features = ["tcp"] }
```

tesuji replied:
Also tokio:
```diff
-tokio = { version = "0.2" }
+tokio = { version = "0.2.2", default-features = false, features = ["tcp", "sync"] }
```

tetsuharuohzeki replied:
These dependencies are for testing. I seem we don't have to specify them strictly.

tesuji replied:
At least for me it will limit the unused indirection deps that we use.

tetsuharuohzeki replied:
Okay. I fixed them in https://github.com/rust-lang/rustup/pull/2154/commits/0f3af10c8b9b3eacbe9a404edd250d9d15f66343

tesuji commented on download/Cargo.toml:
```diff
@@ -16,11 +16,11 @@ reqwest-backend = ["reqwest", "env_proxy", "lazy_static"]
+env_proxy = { version = "0.4", optional = true }
 lazy_static = { version = "1.0", optional = true }
-reqwest = { version = "0.9.21", features = ["socks"], optional = true }
+reqwest = { git = "https://github.com/seanmonstar/reqwest", rev = "7631c0390e90d539b825ed27693a7a0bb1e31dc8", features = ["blocking", "gzip"], optional = true }
```
Make this a new table:
```toml
[dependencies.reqwest]
...
```

tetsuharuohzeki committed:
Upgrade reqwest to v0.10 to parepare async/.await

reqwest v0.9 exposed blocking API as `reqwest::*`.
However, v0.10 expose async API by default and blocking APIs are exposed
as `reqwest::blocking`.

By this change, we migrate to `reqwest::blocking` only.

If we change our codes to async/.await, I think we should do it in
an another changeset because async-fy might require large design change.

tetsuharuohzeki committed:
Update Cargo.lock

tetsuharuohzeki commented:
reqwest v0.10 [has been released](https://github.com/seanmonstar/reqwest/releases/tag/v0.10.0). I'll update this pull request.

tetsuharuohzeki commented:
CI is nice.

tetsuharuohzeki committed:
Turn off some unused features in testings

tetsuharuohzeki commented:
CI is nice and I have addressed all points.

kinnison reviewed (approved):
LGTM, thank you for your work.
